packages/components/panel/src/panel.scss

```diff
@@ -1,104 +1,136 @@
-  justify-content: space-between;
+:host([collapsible]) {
+  [part='header'] {
+    grid-template-columns: auto minmax(auto, 1fr) minmax(40px, auto);
+  }
+}
 
-  > * {
-    // FIXME: This is a quick hack to vertically align the heading with any action buttons
-    padding-block-start: 2px;
+:host([collapsible][toggle-placement='end']) {
+  [part='header'] {
+    grid-template-columns: 1fr minmax(0, auto) auto;
   }
 }
 
-[part='wrapper'] {
-  align-items: center;
+:host([no-padding]) {
+  [part='content'] {
+    padding: 0;
+  }
+}
+
+:host([subtitle-placement='top']) {
+  [part='titles'] {
+    flex-direction: column-reverse;
+  }
+}
+
+:host([no-header]) {
+  [part='header'] {
+    padding: 0;
+  }
+}
+
+[part='titles'] {
   display: flex;
-  gap: 0.5rem;
-  padding-block: var(--_padding-block);
-  padding-inline: calc(var(--_padding-inline) + var(--_wrapper-padding-inline));
+  flex-direction: column;
+  gap: var(--sl-space-new-sm);
 }
 
-button {
-  appearance: none;
-  background: transparent;
-  border: 0;
-  border-radius: var(--sl-border-radius-button-md);
-  cursor: pointer;
+[part='header'] {
+  align-items: center;
+  display: grid;
   flex: 1;
-  outline: 0;
-  padding: 0;
-  text-align: start;
+  gap: var(--sl-space-new-lg);
+  grid-template-columns: 1fr minmax(0, auto);
+  justify-content: space-between;
+  padding: var(--sl-space-new-lg);
+}
 
-  &:focus-visible {
-    outline: var(--_focus-outline);
-    outline-offset: var(--_focus-outline-offset);
-  }
+[part='wrapper'] {
+  align-items: center;
+  display: flex;
+  flex-wrap: wrap;
+  gap: var(--sl-space-new-lg);
 }
 
-sl-icon {
-  rotate: 180deg;
-  transition: rotate 0.2s ease-in-out;
+.hidden {
+  display: none;
 }
 
 slot[name='heading'],
 slot[name='heading']::slotted(*) {
-  color: var(--sl-color-accordion-default-foreground);
-  font: var(--sl-text-accordion-title);
+  color: var(--sl-color-text-bold);
+  font-size: 1.5em;
+  line-height: 1;
 }
 
 slot[name='heading'] {
   display: block;
 }
 
-sl-tool-bar {
-  min-inline-size: 0;
-
-  &:not([empty]) {
-    padding-inline-end: var(--_padding-inline);
-  }
+slot[name='subtitle'],
+slot[name='subtitle']::slotted(*) {
+  color: var(--sl-color-text-bold);
+  line-height: 1.5;
 }
 
-slot[name='actions']::slotted(:first-child) {
-  // FIXME: Since the `aside` slot always has content, gap on the header won't work as expected
-  // So we explicitly set a left margin on the first action
-  margin-inline-start: 1rem;
+sl-tool-bar {
+  min-inline-size: 0;
 }
 
 [part='body'] {
   display: grid;
   grid-template-rows: 1fr;
-  transition: grid-template-rows 0.2s ease-in-out;
+
+  @media (prefers-reduced-motion: no-preference) {
+    transition: grid-template-rows 300ms ease-in-out;
+  }
 }
 
 [part='inner'] {
   overflow: hidden;
 }
 
 [part='content'] {
-  padding-block-end: var(--_padding-block);
-  padding-inline: calc(var(--_padding-inline) + var(--_wrapper-padding-inline));
+  padding: var(--sl-space-new-lg);
+}
+
+.icon {
```

Comment:
There is only 1 `<sl-icon>` in the shadow DOM, right? So you can just use the element selector and remove the `icon` class?